### ISSUE TITLE
Changes in basic group descriptions are ignored

### ISSUE THREAD

**Issue by ars3niy** [tdlib/td#1076]
Steps to reproduce:

1. Log in with tdlib
2. Log in with another client on the same account
3. Create a basic group
4. Change description of the basic group with that other client

Actual result:
* tdlib sends an updateBasicGroupFullInfo with empty description

Expected result
* tdlib sends an updateBasicGroupFullInfo with description set to the new value

tdlib log covering step 4: [updatedGroupDescriptionsCleaned.log](https://github.com/tdlib/td/files/4708078/updatedGroupDescriptionsCleaned.log)

**Comment by ars3niy** [comment]
To mention, one situation where correct description is returned is this:
1. Log in with another client
2. Create a basic group
3. Change basic group description
4. Log in with tdlib

[newGroupWithDescAtLoginCleaned.log](https://github.com/tdlib/td/files/4708082/newGroupWithDescAtLoginCleaned.log)

**Comment by levlam** [comment]
@ars3niy Everything works as expected. FullInfo doesn't synchronized automatically between clients (this is the reason of FullInfo existence). You need to explicitly call `getBasicGroupFullInfo` to receive actual value. Even in that case TDLib will return the cached value first and send `updateBasicGroupFullInfo` after it would be able to connect with the server and update it.

**Comment by ars3niy** [comment]
But it seems that all subsequent getBasicGroupFullInfo requests (executed while online and connected) yield empty description.

Creating basic group and setting description in another client: [groupDescriptionTest3Cleaned.log](https://github.com/tdlib/td/files/4708158/groupDescriptionTest3Cleaned.log)

getBasicGroupFullInfo request (after restarting tdlib client): [groupDescriptionTest3NextLoginCleaned.log](https://github.com/tdlib/td/files/4708159/groupDescriptionTest3NextLoginCleaned.log)

**Comment by levlam** [comment]
This is also true. The description will be updated only after you receive the basic group again from the server. For example, this happens if someone sends a non-text message there, or a text message while the client isn't run.

**Comment by ars3niy** [comment]
> For example, this happens if someone sends a non-text message there

And it looks like it must be someone other than me – and necessarily while tdlib client is not running: people posting photos, stickers and animations while tdlib client is running doesn't seem to make any difference.

**Comment by levlam** [comment]
After a non-text message is received in the basic group, a call to `getBasicGroupFullInfo` should lead to `updateBasicGroupFullInfo` with the updated description. At least, this is supposed to work.

**Comment by ars3niy** [comment]
Say, this is what happens when I send picture as a photo in another client (from same account), then send picture as a document (from another client, same account), then trigger getBasicGroupFullInfo request in the tdlib client: [SendPictureThenGetFullInfoCleaned.log](https://github.com/tdlib/td/files/4708605/SendPictureThenGetFullInfoCleaned.log)

There is no updateBasicGroupFullInfo after getBasicGroupFullInfo request, just the response right away and then nothing more (also tried changing group description before sending the files, same result).

**Comment by ars3niy** [comment]
Another example: another user sending a sticker while tdlib client is offline, then logging in with tdlib and performing getBasicGroupFullInfo: 
```
[1;36m[ 3][t 4][1591083260.865543127][Td.cpp:3334][#4][!Td][&td_requests]	Receive request 11: getBasicGroupFullInfo {
  basic_group_id = 465288209
}[0m
[1;36m[ 3][t 4][1591083260.865580559][ContactsManager.cpp:8294][#4][!GetGroupFullInfoRequest]	Trying to load full basic group 465288209 from database[0m
[1;36m[ 3][t 4][1591083260.865714073][ContactsManager.cpp:8242][#4][!GetGroupFullInfoRequest]	Successfully loaded full basic group 465288209 of size 52 from database[0m
[1;36m[ 3][t 4][1591083260.865743399][ContactsManager.cpp:12817][#4][!GetGroupFullInfoRequest]	Update administrators in chat -465288209 to {DialogAdministrator[user 1037778500, title = , is_owner = true]}[0m
[1;36m[ 3][t 4][1591083260.865767241][ContactsManager.cpp:12835][#4][!GetGroupFullInfoRequest]	Save administrators of chat -465288209 to database[0m
[1;36m[ 3][t 4][1591083260.865798712][Td.cpp:4459][#4][!Td][&td_requests]	Sending update: updateBasicGroupFullInfo {
  basic_group_id = 465288209
  basic_group_full_info = basicGroupFullInfo {
    description = ""
```
skip
```
[1;36m[ 3][t 4][1591083260.865850210][Td.cpp:4474][#4][!Td][&td_requests]	Sending result for request 11: basicGroupFullInfo {
  description = ""
```
Full log: [BenSendStickerThenGetFullInfoCleaned.log](https://github.com/tdlib/td/files/4715670/BenSendStickerThenGetFullInfoCleaned.log)

**Comment by ars3niy** [comment]
It looks like the code path for updating basic group description is this

```
GetGroupFullInfoRequest
ContactsManager::get_chat_full(ChatId chat_id, bool force, Promise<Unit> &&promise)
    if get_chat_full_force return NULL or is_chat_full_outdated returns true
ContactsManager::send_get_chat_full_query
GetFullChatQuery
GetFullChatQuery::on_result
ContactsManager::on_get_chat_full
    if (chat->description != chat_full->about_) {
      chat->description = std::move(chat_full->about_);
      chat->is_changed = true;
    }
```
There would be a log message "Get full basic group NNNNNNNNN from get_chat_full" or "Get full basic group NNNNNNNNN from get expired chat_full". I've seen this happen when a new basic group is created or joined, or when other people join or leave, but never when messages are posted to the group following description change.

**Comment by levlam** [comment]
Yes, it seems that currently basic group description is reloaded only after chat member list, default permissions or pinned message are changed.

Without server-side API changes this can be fixed only by disabling the cache of `basicGroupFullInfo` at the expence of increased network usage by adding at https://github.com/tdlib/td/blob/e2fd1c13e9a0fbb21aa2992b7b2b2b4a7e66204a/td/telegram/ContactsManager.cpp#L12035 something like
```
    if (!force) {
      send_get_chat_full_query(chat_id, std::move(promise), "get expired chat_full");
    } else {
      promise.set_value(Unit());
    }
    return force;
```

**Comment by levlam** [comment]
@ars3niy I have just noticed, that calling `openChat` for the basic group chat also forces its description to be reloaded. Syncronization of group description on chat opening looks like a correct and expected behavior. Do you call `openChat`/`closeChat` whenever a chat is opened/closed?

**Comment by ars3niy** [comment]
Thanks, I'll try. I saw it was written somewhere that you were supposed to call openChat but there didn't seem to be any problems. This and the fact that avatars are a bit slow to update could be the only consequences.
